scripts/global.js

```diff
@@ -0,0 +1,1181 @@
+    clearInterval(vanUiUpdater);
+}
+
+export async function generateVanityWallet() {
+    if (isVanityGenerating) return stopSearch();
+    if (typeof Worker === 'undefined')
+        return createAlert('error', ALERTS.UNSUPPORTED_WEBWORKERS, [], 7500);
+    // Generate a vanity address with the given prefix
+    if (
+        doms.domPrefix.value.length === 0 ||
+        doms.domPrefix.style.display === 'none'
+    ) {
+        // No prefix, display the intro!
+        doms.domPrefix.style.display = 'block';
+        doms.domGenKeyWarning.style.display = 'none';
+        doms.domGuiAddress.innerHTML = '~';
+        doms.domPrefix.focus();
+    } else {
+        // Remove spaces from prefix
+        doms.domPrefix.value = doms.domPrefix.value.replace(/ /g, '');
+
+        // Cache a lowercase equivilent for lower-entropy comparisons (a case-insensitive search is ALOT faster!) and strip accidental spaces
+        const nInsensitivePrefix = doms.domPrefix.value.toLowerCase();
+        const nPrefixLen = nInsensitivePrefix.length;
+
+        // Ensure the input is base58 compatible
+        for (const char of doms.domPrefix.value) {
+            if (!MAP_B58.toLowerCase().includes(char.toLowerCase()))
+                return createAlert(
+                    'warning',
+                    ALERTS.UNSUPPORTED_CHARACTER,
+                    [{ char: char }],
+                    3500
+                );
+            // We also don't want users to be mining addresses for years... so cap the letters to four until the generator is more optimized
+            if (doms.domPrefix.value.length > 5)
+                return createAlert(
+                    'warning',
+                    ALERTS.UNSUPPORTED_CHARACTER,
+                    [{ char: char }],
+                    3500
+                );
+        }
+        isVanityGenerating = true;
+        doms.domPrefix.disabled = true;
+        let attempts = 0;
+
+        // Setup workers
+        const nThreads = Math.max(
+            Math.floor(window.navigator.hardwareConcurrency * 0.75),
+            1
+        );
+        console.log('Spawning ' + nThreads + ' vanity search threads!');
+        while (arrWorkers.length < nThreads) {
+            arrWorkers.push(
+                new Worker(new URL('./vanitygen_worker.js', import.meta.url))
+            );
+            arrWorkers[arrWorkers.length - 1].onmessage = (event) =>
+                checkResult(event.data);
+            arrWorkers[arrWorkers.length - 1].postMessage(
+                cChainParams.current.PUBKEY_ADDRESS
+            );
+        }
+
+        // GUI Updater
+        doms.domVanityUiButtonTxt.innerText =
+            'Stop (Searched ' + attempts.toLocaleString('en-GB') + ' keys)';
+        vanUiUpdater = setInterval(() => {
+            doms.domVanityUiButtonTxt.innerText =
+                'Stop (Searched ' + attempts.toLocaleString('en-GB') + ' keys)';
+        }, 200);
+
+        function checkResult(data) {
```

Comment:
Same here with es-lint added no inner function declarations can be made, we need to fix this portion to its own function and being called here